spring-kafka/src/main/java/org/springframework/kafka/listener/RecoveryStrategy.java

```diff
@@ -35,10 +36,11 @@
 	 * @param record the record.
 	 * @param ex the exception.
 	 * @param container the container (or parent if a child container).
+	 * @param consumer the consumer.
 	 * @return true to skip.
 	 * @throws InterruptedException if the thread is interrupted.
 	 */
-	boolean recovered(ConsumerRecord<?, ?> record, Exception ex, @Nullable MessageListenerContainer container)
-			throws InterruptedException;
+	boolean recovered(ConsumerRecord<?, ?> record, Exception ex, @Nullable MessageListenerContainer container,
+			@Nullable Consumer<?, ?> consumer) throws InterruptedException;
```

Comment:
Looks like a breaking change in the public API signature...

Reply:
It's not public; it's package-private.

Reply:
Oops - sorry - no - this interface is new in 2.7.